bin/clover/src/spec/props.ts

```diff
@@ -576,6 +576,7 @@ export type DefaultPropType = "domain" | "secrets" | "resource_value";
 export function createObjectProp(
   name: string,
   parentPath: string[],
+  cfProp: CfObjectProperty | undefined,
```

Comment:
What's the advantage of declaring cfProp this way instead of `cfProp?: CfObjectProperty`?

Reply:
Oh, I did it this way so that typescript would yell at me anywhere I tried to call it without a createObjectProp()--it forced me to make a decision in each place we create an object prop. It's not as necessary now that that initial turn is done, but doesn't hurt to show the explicit decision in the code IMO.

Reply:
i.e. if you don't use `?`, then you can pass `undefined` but you can't omit the parameter.